Hugo Modules Guide (#266)
CloudCannon/platform-documentation: guides/hugo-modules/modules-with-site-mounting.mdx

In Site Mounting, CloudCannon refers to the donor site as the *Remote Site* and the site that receives files as the *Local Site*. In this example, the remote site will contain your modules. The local site is your main site.

<comp.Notice info_type="important">
  This setup requires that you own your site repository and the repositories for your modules. These repositories should be in the same GitHub organization. If you use a public repository owned by someone else, you must fork it into your GitHub organization.
</comp.Notice>

## Disable the default Hugo Module import

To emulate Hugo Modules with Site Mounting, you must disable the default Hugo Module behavior in CloudCannon. The simplest method of disabling this behavior is to create a second Hugo configuration file for use in CloudCannon, which does not contain your module import.

Your configuration file will look something like this:

<comp.CodeBlock language="toml" source="hugo.cloudcannon.toml">
```
baseURL = 'https://example.com/'

# ... all other config

# module.imports commented out or removed
# [[module.imports]]
# path = 'github.com/my/module'

```
</comp.CodeBlock>

> <comp.Notice info_type="info">
>   If your site uses a <a rel="noopener noreferrer" href="https://gohugo.io/getting-started/configuration/">configuration directory</a> for Hugo, adding a `cloudcannon` environment is another option.
> </comp.Notice>

Reviewer: ```suggestion
<comp.Notice info_type="info">
  If your site uses a <a rel="noopener noreferrer" href="https://gohugo.io/getting-started/configuration/">configuration directory</a> for Hugo, another option is to add a `cloudcannon` environment.
</comp.Notice>
```

By default, these files are accessible on this site's <a href="https://cloudcannon.com/documentation/articles/viewing-your-site-on-a-testing-domain/" rel="noreferrer">.cloudvent.net testing domain</a>. To prevent access to the testing domain, switch the module site to <a href="https://cloudcannon.com/documentation/articles/adding-password-authentication-to-your-site/" rel="noreferrer">password authentication</a> and do not set a password.

## Mount the module site to your main site

Open your main site in CloudCannon. To configure your Site Mounting settings, navigate to the *Site Mountings* page under *Site Settings*, then *Builds*.

<comp.DocsImage path="https://cc-dam.imgix.net/documentation/images/guides/hugo-modules/2024-03/CloudCannon-Documentation-Site-Settings-Site-Mounting.png" alt="A screenshot of the CloudCannon app shows the Site Mountings page, with a section to add a mounted site to the local site." title="Add a mounted site" type="screenshot"/>

In the *Remote site* field, select the CloudCannon site that contains your module files.

Leave the *Remote artifact* field set to "Source files".

> Enter the paths you want your module to mount for your *Remote* and *Local paths*. For example:

Reviewer: ```suggestion
Enter the paths you want your module to mount for your *Remote* and *Local paths*. For example, given the following Hugo configuration:
```

<comp.CodeBlock language="yaml" source="module/config.yml">
```
module:
  mounts:
    - source: theme/layouts
      target: layouts
```
</comp.CodeBlock>

If your module configuration were the code above, you would enter "theme/layouts" into the *Remote path* field and "layouts" in the *Local path* field.

Once you have completed your configuration, click the *Mount site* button.

## New build of your main site

In your main site on CloudCannon, navigate to the *Status* page. Click the *Rebuild* button. CloudCannon will mount the module content at the your specified path before running your Hugo build command.

Any changes to your module repository will trigger a new build of any site to which the content is mounted. You don't have to worry about keeping your sites up to date.

## Mounting multiple paths

> In some cases, you may want to mount your module to multiple paths. For example:

Reviewer: ```suggestion
In some cases, you may want to mount your module to multiple paths. For example, given the following Hugo configuration:
```
